### PR TITLE
[HWORKS-791] Add Python API support to schedule jobs

### PR DESCRIPTION
**Pull request by SirOibaf**

### CODE REVIEW

File: python/hopsworks/job.py [hunk 1]
```diff
@@ -47,6 +50,11 @@ def __init__(
         self._creator = creator
         self._executions = executions
         self._project_id = project_id
+        self._job_schedule = (
+            js.JobSchedule.from_response_json(job_schedule)
+            if job_schedule
```
Comment (robzor92) [inline python/hopsworks/job.py:55]:
```suggestion
            if job_schedule is not None
```

File: python/hopsworks/job.py [hunk 2]
```diff
@@ -180,6 +193,49 @@ def delete(self):
         """
         self._job_api._delete(self)
 
+    def schedule(self, cron_expression, start_time=None, end_time=None):
```
Comment (robzor92) [inline python/hopsworks/job.py:196]: To be consistent with how we show what datatypes in the docs for client facing APIs - Can you remove the `datetime, optional` etc from the docstring below and instead add the type in the function signature? I.e like this https://github.com/logicalclocks/hopsworks-api/blob/14f46bca54f4f8f5eedd5c32d5cd3b86be590b17/python/hopsworks/job.py#L120

The resulting docs will show the type for each function parameter: https://docs.hopsworks.ai/hopsworks-api/3.4/generated/api/jobs/#run

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: python/hopsworks/job.py (hunk 1 above)
Comment (SirOibaf) [reply python/hopsworks/job.py:55]: In Python `if job_schedule` will return true if the object is not None or empty string.

File: python/hopsworks/job.py (hunk 2 above)
Comment (SirOibaf) [reply python/hopsworks/job.py]: Fixed

**Review by robzor92 — APPROVED** [review]

**Pull request merged**